src/stdplus/tab.hpp

```diff
@@ -141,13 +156,13 @@ typename T2::iterator copy_safe_apart(size_t n, const T1 & tab_src, T2 & tab_dst
 
 		template <typename TC> // TC = type of some container. it must be linear-memory. TODO: metaprogramming check
 		tab_range(const TC & obj, size_t begin, size_t end)
-				: m_begin(&(obj.at(begin))), m_end(&(obj.at(end-1))+1), m_size(end-begin) {
+				: m_throw_if_less_init_list(end, begin), m_begin(&(obj.at(begin))), m_end(&(obj.at(end-1))+1), m_size(end-begin) {
```

Comment:
note:  end is size_t.  If it would be a pointer then it would be UB:


or simply, calculating  end-1 where end is the .end() of some container,
if that container is empty,
is UB:

c++17 draft http://eel.is/c++draft/expr.add 8.7
end points to x[n] (according to footnote 86)
P points to x[n] (footnote 86)
P points to x[i]
n==0 (because empty array)
i==0
so we calculate "end-1"
  where end == emptyarray.end()
operator+ (8.7/4) works ok if:
"if 0≤i+j≤n; otherwise, the behavior is undefined"
i==0 , j==-1.
0 <= i+j <= n  should be true to not have UB
0 <= 0-1 <= n  should be true to not have UB
0 <= -1 <= 0  should be true to not have UB
this is false
so this is UB